seqr/views/utils/dataset_utils.py

```diff
@@ -413,19 +407,20 @@ def _load_rna_seq(model_cls, file_path, user, mapping_file, ignore_extra_samples
         sample_project_tuples=samples_by_id.keys(),
         data_source=data_source,
         sample_id_to_individual_id_mapping=sample_id_to_individual_id_mapping,
-        raise_unmatched_error_template=None if ignore_extra_samples else 'Unable to find matches for the following samples: {sample_ids}'
+        raise_unmatched_error_template=None if ignore_extra_samples else 'Unable to find matches for the following samples: {sample_ids}',
+        sample_id_to_tissue_type=sample_id_to_tissue_type,
     )
 
-    warnings = []
-    if validate_samples:
-        samples = validate_samples(samples, sample_id_to_tissue_type, warnings)
-
     # Delete old data
     to_delete = model_cls.objects.filter(sample__in=samples).exclude(sample__data_source=data_source)
     prev_loaded_individual_ids = set(to_delete.values_list('sample__individual_id', flat=True))
     if to_delete:
         model_cls.bulk_delete(user, to_delete)
 
+    samples_to_update = [s.id for s in samples if s.data_source != data_source]
+    if samples_to_update:
+        Sample.bulk_update(user, {'data_source': data_source}, id__in=samples_to_update)
```

Comment:
Can some of the existing samples have the same data source while loading the same data file multiple times while fixing errors in the file?

Reply:
its possible but its a pretty rare edge case and in that case doing an extra bulk update on those samples wouldn't hurt

Reply:
The `samples` for the existing samples is not a `queryset` because we remove project and tissue mismatched models from the queryset. The easiest way to create a queryset from the `samples` is to filter again with the `id__in={s.id for s in samples}` condition.

Reply:
ah I forgot about that, good catch